kani-compiler/src/kani_middle/attributes.rs

```diff
@@ -185,29 +294,65 @@ impl<'tcx> KaniAttributes<'tcx> {
     /// Note that all attributes should be valid by now.
     pub fn harness_attributes(&self) -> HarnessAttributes {
         // Abort if not local.
-        assert!(self.item.is_local(), "Expected a local item, but got: {:?}", self.item);
+        let Some(local_id) = self.item.as_local() else {
+            panic!("Expected a local item, but got: {:?}", self.item);
+        };
         trace!(?self, "extract_harness_attributes");
         assert!(self.is_harness());
-        self.map.iter().fold(HarnessAttributes::default(), |mut harness, (kind, attributes)| {
-            match kind {
-                KaniAttributeKind::ShouldPanic => harness.should_panic = true,
-                KaniAttributeKind::Solver => {
-                    harness.solver = parse_solver(self.tcx, attributes[0]);
-                }
-                KaniAttributeKind::Stub => {
-                    harness.stubs = parse_stubs(self.tcx, self.item, attributes);
-                }
-                KaniAttributeKind::Unwind => {
-                    harness.unwind_value = parse_unwind(self.tcx, attributes[0])
-                }
-                KaniAttributeKind::Proof => harness.proof = true,
-                KaniAttributeKind::Unstable => {
-                    // Internal attribute which shouldn't exist here.
-                    unreachable!()
-                }
-            };
-            harness
-        })
+        let mut attrs = self.map.iter().fold(
+            HarnessAttributes::default(),
+            |mut harness, (kind, attributes)| {
+                match kind {
+                    KaniAttributeKind::ShouldPanic => harness.should_panic = true,
+                    KaniAttributeKind::Solver => {
+                        harness.solver = parse_solver(self.tcx, attributes[0]);
+                    }
+                    KaniAttributeKind::Stub => {
+                        harness.stubs = parse_stubs(self.tcx, self.item, attributes);
+                    }
+                    KaniAttributeKind::Unwind => {
+                        harness.unwind_value = parse_unwind(self.tcx, attributes[0])
+                    }
+                    KaniAttributeKind::Proof | KaniAttributeKind::ProofForContract => {
+                        harness.proof = true
+                    }
+                    KaniAttributeKind::Unstable => {
+                        // Internal attribute which shouldn't exist here.
+                        unreachable!()
+                    }
+                    KaniAttributeKind::CheckedWith
+                    | KaniAttributeKind::ReplacedWith
+                    | KaniAttributeKind::IsContractGenerated => {
+                        todo!("Contract attributes are not supported on proofs")
+                    }
+                };
+                harness
+            },
+        );
+        let current_module = self.tcx.parent_module_from_def_id(local_id);
```

Comment:
Can we do this when we match the `KaniAttributeKind::ProofForContract` instead? Can you please also export this to a function that returns the stub pair?

Reply:
I could but then I need to change the handling of `stub`, because that overrides `attrs.stubs`. If you're ok with that I can move it. 

Beyond that the main reason this is extra is because we will need to treat replacement in essentially the same way. In particular the second `map` is common. If I refactor it into a function for `proof_for_contract` it won't be obvious that the code is the same. 

I could factor the whole "get stubbing for contracts stuff" into a separate function though if you like without obfuscating the shared code?